rotkehlchen/serialization/deserialize.py

```diff
@@ -380,24 +381,36 @@ def deserialize_trade_pair(pair: str) -> TradePair:
     return TradePair(pair)
 
 
-def deserialize_asset_movement_category(symbol: str) -> AssetMovementCategory:
+def deserialize_asset_movement_category(
```

Comment:
I can see why this would be useful when we use these `HistoryEvent` objects everywhere but perhaps for now it does not make much sense to add it in this funciton since it's only used in Kraken and we end up burdening every other place where this is called?

Reply:
What do you mean? Does it have any negative impact?

Reply:
Well it makes the deserialization take more time since it has more logic.  Adds the extra step to check for HistoryEventType for every place where deserialize_asset_movement_category() is called

Reply:
I refactored the function so it exists earlier if is str and less checks are made if it is a str

Reply:
Hmm still not liking this but will let it pass for now